### PR TITLE
added validation to user profile setup

### PR DESCRIPTION
**Pull request by Pauullamm**
#### Summary
This PR introduces additional validation logic and enhances the user profile setup functionality by ensuring that social media links are correctly provided before storing user data in Firestore.

#### Changes Introduced
- **Social Media Links Validation:**
  - Added a validation check to ensure that the `socialLinks` field is an object and not null.
  - Implemented a check to verify that at least one social media link is filled in (i.e., non-empty) before proceeding. If all social links are empty, an error is logged, and an exception is thrown to prevent further execution.
  
- **Firestore User Document Update:**
  - Integrated the validated `socialLinks` into the user document stored in Firestore.
  - Ensured that the user document includes `socialLink`, `fullName`, `occupation`, `bio`, and `profilePhotoUrl` (which was not added previously).

#### Logging Enhancements
- Added logging to capture error scenarios, such as missing social media links or when no social link is filled, to aid in debugging and tracking issues related to user profile updates.

#### Testing & Verification
- Tested the feature with various scenarios, including providing no social links, one valid link, and multiple valid links, to ensure the validation logic works as expected using Postman.
- Verified that valid user profiles are correctly stored in Firestore, and the onboarding status progresses as intended.

#### Impact
This update will prevent users from proceeding with an incomplete profile setup, ensuring data integrity in the Firestore database and improving the user experience by providing clear error messages for missing or invalid input.

---

Please review the changes and let me know if there are any questions or further improvements needed.

### CODE REVIEW

File: functions/controllers/onboardingController.js [hunk 1]
```diff
@@ -295,7 +303,29 @@ const setupUserProfile = functions.https.onCall(async (data, context) => {
...
+      if (!hasAtLeastOneLink) {
+        logger.error(
+          'Error updating user profile, at least one social media link is required'
+        );
+        throw new functions.https.HttpsError(
+          'internal',
+          'no social media links provided'
+        );
+      }
+    }
```
Comment (Pauullamm) [inline functions/controllers/onboardingController.js:320]: I did not add validation to ensure no duplicate keys are provided e.g. 
-  facebook: link
- facebook: link
- twitter: link

As the preceding key-value pair is ignored automatically in javascript

File: functions/controllers/onboardingController.js [hunk 2]
```diff
@@ -304,15 +334,20 @@ const setupUserProfile = functions.https.onCall(async (data, context) => {
...
       bio: bio,
+      profilePhotoUrl: profilePhotoUrl,
     };
 
     logger.log('Updating user document in Firestore', userDoc);
 
     await userRef.set(userDoc, { merge: true });
-
+    // Call the Onboarding Status Transition Utility function to advance the status
+    const onboardingResult = await progressOnboardingStatus(uid);
```
Comment (Pauullamm) [inline functions/controllers/onboardingController.js:335]: Added this call to advance onboarding status, which I did not do previously sorry!

File: functions/controllers/onboardingController.js [hunk 3]
```diff
@@ -304,15 +334,20 @@ const setupUserProfile = functions.https.onCall(async (data, context) => {
       occupation: occupation,
       socialLink: socialLinks,
       bio: bio,
+      profilePhotoUrl: profilePhotoUrl,
```
Comment (Pauullamm) [inline functions/controllers/onboardingController.js]: Added key-value pair for firebase storage url

File: functions/controllers/onboardingController.js [hunk 4]
```diff
@@ -277,10 +277,18 @@ const advanceOnboardingStatus = functions.https.onCall(
...
 
     // Validate fields
-    if (!uid || !fullName || !occupation || !socialLinks || !bio) {
+    if (
+      !uid ||
+      !fullName ||
+      !occupation ||
+      !socialLinks ||
+      !bio ||
+      !profilePhotoUrl
```
Comment (ademiltonnunes) [inline functions/controllers/onboardingController.js]: Sorry for not mentioning it. profilePhotoURL doesn't need to be required, you can remove this validation

**Review by ademiltonnunes — COMMENTED** [review]
Good job overall. Only a modification was reported.
As soon as you complete it, I will prove your PR.

**Review by ademiltonnunes — APPROVED** [review]
Good job. Congrats!

**Pull request merged**